cmake/SetupCCache.cmake

```diff
@@ -5,10 +5,21 @@ option(USE_CCACHE "Use CCache if available of speed up builds" ON)
 if(USE_CCACHE)
   find_program(CCACHE_FOUND ccache)
   if(CCACHE_FOUND)
+    string(CONCAT CCACHE_COMMAND
+      "CCACHE_SLOPPINESS=pch_defines,time_macros "
+      # In order to make ccache work with precompiled headers we need to:
+      # - Hash the content of the compiler rather than the location and mtime
+      # - Hash the header file in the repo that will generate the precompiled
+      #   header
+      # - Ignore the precompiled header in the build directory
+      "CCACHE_COMPILERCHECK=content "
```

Comment:
Okay, so I'm okay with this if we only add the `content` check on macOS and CI and use the default on Linux